tests/pipeline/test_course_daily_metrics.py

```diff
@@ -91,37 +96,43 @@ def setup(self, db):
             self.course_enrollments = [CourseEnrollmentFactory(
                 course=self.course_overview) for i in range(4)]
 
-        if organizations_support_sites():
-            self.my_site = SiteFactory(domain='my-site.test')
-            self.my_site_org = OrganizationFactory(sites=[self.my_site])
-            OrganizationCourseFactory(organization=self.my_site_org,
-                                      course_id=str(self.course_overview.id))
-            for ce in self.course_enrollments:
-                UserOrganizationMappingFactory(user=ce.user,
-                                               organization=self.my_site_org)
+        self.site = SiteFactory(domain='my-site.test')
+        self.edly_sub_organization = EdlySubOrganizationFactory(
+            lms_site=self.site
+        )
+        OrganizationCourseFactory(
+            organization=self.edly_sub_organization.edx_organization,
+            course_id=str(self.course_overview.id)
+        )
+        for course_enrollment in self.course_enrollments:
+            EdlyUserProfileFactory(
+                user=course_enrollment.user,
+                edly_sub_organizations=[self.edly_sub_organization]
+            )
 
         self.course_access_roles = [CourseAccessRoleFactory(
             user=self.course_enrollments[i].user,
             course_id=self.course_enrollments[i].course_id,
             role=role,
+            org=self.course_enrollments[i].course_id.org,
             ) for i, role in enumerate(self.COURSE_ROLES)]
 
         # create student modules for yesterday and today
-        for day in [prev_day(self.today), self.today]:
```

Comment:
This loop creates duplicates the enrollments for no particular reason violating the `unique_together` constraint(https://github.com/edly-io/edx-platform/blob/cfb53bb9bba9af69d7eac7e2ba86ed45bcde686b/lms/djangoapps/courseware/models.py#L106) for course and student so I removed the loop.

Reply:
@ziafazal  It raised the error when I ran the tests for the first time. I don't know how it was working for appsembler. or even if the tests were working even. Most of the changes in these tests are at least a year old.